### ISSUE TITLE
NavDrawerViewModel leaked

### ISSUE THREAD

**Issue by p26432** [JohannBlake/Jetmagic#4]
LeakCanary reports that an instance of NavDrawerViewModel is leaked after ViewModel.onCleared() is invoked.

**Comment by JohannBlake** [comment]
onCleared is not used anywhere in the app.

**Comment by JohannBlake** [comment]
The same comments for issue [https://github.com/JohannBlake/Jetmagic/issues/3](https://github.com/JohannBlake/Jetmagic/issues/3)  apply to this issue as well.

**Comment by p26432** [comment]
Subclasses of ViewModel() may optionally override onCleared() to perform
extra shutdown tasks. The default onCleared() just does a super.clear().

LeakCanary stopped complaining about a leaked instance of MainViewModel()
when I made the static variable mainViewModel nullifiable and nulled it out
in an override of onCleared() I added to MainViewModel(). Of course, I also
had to change the references to mainViewModel to make them null-safe.
